app/classes/inat/obs.rb

```diff
@@ -0,0 +1,463 @@
+# frozen_string_literal: true
+
+#
+#  = Inat::Obs Object
+#
+#  Represents the result of an iNat API search for one observation,
+#  mapping iNat key/values to MO Observation attributes
+#
+#  == Class methods
+##
+#  == Instance methods
+#
+#  === iNat attributes & associations
+#
+#  obs::                 The iNat observation data
+#  inat_description::
+#  inat_id::
+#  inat_identifications    array of identifications, taxa need not be unique
+#  inat_location::         lat,lon
+#  inat_obs_fields::       array of fields, each field a hash
+#  inat_obs_photos::       array of observation_photos
+#  inat_place_guess::
+#  inat_private_location:: lat, lon
+#  inat_project_names::
+#  inat_prov_name::        provisional species name
+#  inat_positional_accuracy::  accuracy of inat_location in meters
+#  inat_public_positional_accuracy::  blurred accuracy
+#  inat_quality_grade::    casual, needs id, research
+#  inat_tags::             array of tags
+#  inat_taxon_name::       scientific name
+#  inat_taxon_rank::       rank (can be secondary)
+#  inat_user_login
+#
+#  == MO attributes
+#  gps_hidden
+#  license::
+#  name_id
+#  notes
+#  lat
+#  lng
+#  location
+#  text_name
+#  when
+#  where
+#
+# == Other mappings used in MO Observations
+#
+#  dqa::               data quality grade
+#  provisional_name::  MO text_name corresponding to inat_prov_name
+#
+# == Utilities
+#
+#  importable?::  Is it importable to MO?
+#
+class Inat
+  class Obs
+    def initialize(imported_inat_obs_data)
+      @obs = JSON.parse(imported_inat_obs_data, symbolize_names: true)
+    end
+
+    ########## iNat attributes
+
+    def inat_description
```

Comment:
I'd start by loking for a way of getting rid of this pattern,  This seems very repetitive.